### ISSUE TITLE
Gambatte: cgb in gba doesn't do anything?

### ISSUE THREAD

**Issue by nattthebear** [TASVideos/BizHawk#2100]
Expected to see indication on the title screen of Shantae, saw nothing.

This is a regression, but I wouldn't prioritize it very high; don't think any games do anything interesting with this mode.

**Comment by ThunderAxe31** [comment]
Actually, this is very important, as it's necessary for console verification of all GB and GBC games.

**Comment by nattthebear** [comment]
I didn't bisect, but here's the expected behavior from 1.12.1:

![image](https://user-images.githubusercontent.com/1824709/83882220-ce8e4600-a70f-11ea-90eb-99b89a365640.png)

> Actually, this is very important, as it's necessary for console verification of all GB and GBC games.

I'll get Yoshi to add an even lower Priority tag :+1:

**Comment by TiKevin83** [comment]
This is a pretty serious regression if true as it affects cycle count in console verification, but I'd be very confused how/when it broke

**Comment by nattthebear** [comment]
So, the thing that caused me to investigate this was working on https://github.com/TASVideos/BizHawk/issues/2067 with https://github.com/TASVideos/BizHawk/commit/ad16294b11a4e12c9f41d88aa100210c498bea26

For that, I took the lowest effort approach to fixing the problem:  Just use Sameboy's open source alternate bioses.  And I discovered there that Sameboy in fact has 3 different bioses: regular, color, and advanced.  But we don't seem to have that distinction either in GBHawk or Gambatte, which made me curious to see how 💃 would behave.  That's all the insight I have.

**Comment by TiKevin83** [comment]
https://github.com/TASVideos/BizHawk/pull/1954/commits/99d06d84790aec361db2f9b21518f6e46b96d022

The AGB version of the GBC BIOS has never been dumped, but the assumed contents are inferred from their affects on the RNG and initial registers, so the current system edits the GBC BIOS with the assumed diff.  This is probably not working against the not-real Sameboy BIOS.

**Comment by TiKevin83** [comment]
@nattthebear yep this broke with the casual profile rewiring to not require the real BIOS, the GBC BIOS edit isn't the same for the SameBoy BIOS.

**Comment by nattthebear** [comment]
So I tried a few different roms with a lot of different combinations of bios and other options and apparently I got all jumbled up while doing it.

I opened this bug because I thought the issue was occurring with the real Nintendo bios, but trying it out now I see that it's not, only with the faker bios.  Sorry about the confusion.

Real bios + cgb mode:  OK
Real bios + agb mode:  OK, shows gba enchanced
Faker bios + cgb mode: OK
Faker bios + agb mode: Shows crash as you explained.

**Comment by nattthebear** [comment]
Leaving open but on priority giga poop

**Comment by adelikat** [comment]
Can anyone tell me if this was a regression from 2.4.2? or only broken and fixed since this release?

**Comment by TiKevin83** [comment]
Broken and fixed all within 2.5 development.  Gambatte booting without official BIOS was broken for ~3 years.
